src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -4442,13 +4444,27 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
 
             head.m_v = ASRUtils::EXPR(explicit_iter_var);
         } else {
-            body.reserve(al, x.n_body);
+            body.reserve(al, x.n_body + 2);
             head.m_v = target;
         }
 
         SymbolTable *parent_scope = current_scope;
         current_scope = al.make_new<SymbolTable>(parent_scope);
+        std::string name = "j";
+        ASR::expr_t *j = ASR::down_cast<ASR::expr_t>(ASR::make_Var_t(
+            al, x.base.base.loc, ASR::down_cast<ASR::symbol_t>(ASR::make_Variable_t(
+                al, x.base.base.loc, current_scope, s2c(al, name), nullptr, 0, ASR::intentType::Local,
+                nullptr, nullptr, ASR::storage_typeType::Default,
+                ASRUtils::TYPE(ASR::make_Integer_t(al, x.base.base.loc, 4, nullptr, 0)),
+                ASR::abiType::Source, ASR::accessType::Public,
+                ASR::presenceType::Required, false))));
```

Comment:
Here you have created the Variable, but it is not registered in the current scope.
See: https://github.com/lcompilers/lpython/blob/main/src/libasr/pass/pass_utils.cpp#L503

Reply:
Thanks for letting me know about this function , I've updated my code accordingly . Resolving !